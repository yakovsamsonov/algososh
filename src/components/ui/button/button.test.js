import renderer from 'react-test-renderer';

import { Button } from './button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Кнопка', () => {
  it('содержит текст', () => {
    const buttonWithText = renderer.create(<Button text="Текст" />).toJSON();

    expect(buttonWithText).toMatchSnapshot();
  });

  it('не содержит текст', () => {
    const buttonNoText = renderer.create(<Button />).toJSON();

    expect(buttonNoText).toMatchSnapshot();
  });

  it('неактивна', () => {
    const buttonDisabled = renderer.create(<Button disabled />).toJSON();

    expect(buttonDisabled).toMatchSnapshot();
  });

  it('в состоянии загрузки', () => {
    const buttonDisabled = renderer.create(<Button isLoader />).toJSON();

    expect(buttonDisabled).toMatchSnapshot();
  });

  it('при нажатии вызывает переданную функцию', () => {
    const onClick = jest.fn();

    render(<Button text="Кнопка" onClick={onClick}></Button>);

    const button = screen.getByText('Кнопка');

    fireEvent.click(button);

    expect(onClick).toBeCalledTimes(1);
  });
});
