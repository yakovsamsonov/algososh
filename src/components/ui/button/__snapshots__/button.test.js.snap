// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кнопка в состоянии загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнопка не содержит текст 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка неактивна 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка содержит текст 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;
